synapse/federation/transaction_queue.py

```diff
@@ -714,3 +747,183 @@ def json_data_cb():
             success = False
 
         defer.returnValue(success)
+
+
+@attr.s
+class PerDestinationQueue(object):
+    """
+    Holds the per-destination transmission queues.
+
+    (Currently it only holds the EDUs but at some point it would be good to extend it to
+    hold the other per-destination data like PDUs and presence.)
+    """
+    clock = attr.ib()  # type: synapse.util.Clock
+
+    # the callback to be called when our EDUs become ready for transmission.
+    attempt_transaction_cb = attr.ib()  # type: callable
+
+    # edu buckets for this destination, keyed by edu bucket ID
+    edu_buckets = attr.ib(factory=dict)   # type: dict[int, EduTransmissionBucket]
```

Comment:
Given there are only two buckets, I'd be sorely tempted to just have them as two attributes. It also saves from having to figure out when to create/delete a bucket or not.